src/backends/pgfplotsx.jl

```diff
@@ -1060,23 +1060,22 @@ function pgfx_fillrange_args(fillrange, x, y, z)
     return PGFPlotsX.Coordinates(x_fill, y_fill, z_fill)
 end
 
+sanitize_raw_latex_string(s::AbstractString) =
+    replace(s, r"\\?\#" => raw"\#") |> s -> replace(s, r"\\?\%" => raw"\%")
+
 pgfx_sanitize_string(p::PlotText) = PlotText(pgfx_sanitize_string(p.str), p.font)
 function pgfx_sanitize_string(s::AbstractString)
-    s = replace(s, r"\\?\#" => "\\#")
-    s = replace(s, r"\\?\%" => "\\%")
-    s = replace(s, r"\\?\_" => "\\_")
-    s = replace(s, r"\\?\&" => "\\&")
-    s = replace(s, r"\\?\{" => "\\{")
-    s = replace(s, r"\\?\}" => "\\}")
```

Comment:
Why weren't they tested ?

Reply:
> { and } are breaking the example

Why is that? Latexification should happen after the replacements.
The general philosophy here is, that these characters mean something different in latex source code and thus we escape those such that they show up in the label, say `plot(1:5, xlabel = "_Profit_ of C & A { % of BIP  in $}"`. If one wants to have the latex meaning of those one should use `LaTeXString`s.

Reply:
> so that we can properly justify the missing ones

The following script shows which characters need to be escaped to compile latex without errors:
```julia
for c in ('#', '%', '_', '&', '{', '}', '$')
  @show c
  fn = tempname()
  write(
    fn, """
    \\documentclass{standalone}
    \\begin{document}
      Some text with the char $("\\" * c) in it.
    \\end{document}
    """
  )
  run(`latexmk -quiet -pdf $fn`)
end
```

>  EDIT: Ah.. it's probably in a following call

Yes, `pgfx_sanitize_string` was called twice hence once too much after calling `Latexify` causing spurious escapes. I solved this by returning a `LaTeXString` (without surrounding `$`) instead of a regular `String`.

Reply:
Tests covering these have also been added.

Reply:
> I solved this by returning a LaTeXString

For that we'd need to turn `LaTeXStrings` into a direct dependency. I'd prefer if we can get around of that.
One alternative might be to store the sanitized strings in a different local variable

Reply:
`Latexify` is a direct dependency of Plots https://github.com/JuliaPlots/Plots.jl/blob/6c622e096501559ae81c9c35391c19cf243f556d/Project.toml#L16
`LaTeXStrings` is direct dependency of `Latexify`: https://github.com/korsbo/Latexify.jl/blob/8b860d3710fc0d5374058c0dc0e9d712d81296eb/Project.toml#L10.
So what's wrong in adding `LaTeXStrings` as a direct dependecy for `Plots` ?
I'd prefer that over writing `Latexify.LaTeXString` or worse `Latexify.LaTeXStrings.LaTeXString`.

Reply:
Probably not too bad. We can deletete the whole requires setup for LaTeXStrings then

Reply:
Yes, `@require LaTeXStrings` is already removed in this PR, and `LaTeXStrings` was added to `Project.toml` (I forgot to add compat bounds, will add them now).